src/auth/digest/Config.h

```diff
@@ -25,42 +26,61 @@ class User;
 }
 }
 
-/* Generic */
-typedef struct _digest_nonce_data digest_nonce_data;
-typedef struct _digest_nonce_h digest_nonce_h;
+/* the nonce structure we'll pass around */
+class digest_nonce_h : public hash_link, public RefCountable
+{
+    MEMPROXY_CLASS(digest_nonce_h);
 
-/* data to be encoded into the nonce's hex representation */
-struct _digest_nonce_data {
-    time_t creationtime;
-    uint32_t randomdata;
-};
+public:
+    typedef RefCount<digest_nonce_h> Pointer;
 
-/* the nonce structure we'll pass around */
+    digest_nonce_h() = default;
+    digest_nonce_h(const digest_nonce_h &) = delete; // non-copyable
+    ~digest_nonce_h() { xfree(key); }
+
+    /// The HEX encoded unique identifier for this nonce
+    const char *hex() const { return static_cast<const char *>(key); }
+
+    /// Check the nonce and invalidate if any tests fail.
+    /// \retval true if the nonce is valid.
+    bool valid(char clientCount[9]);
+
+    /// Check the freshness of this nonce and invalidate if stale.
+    /// \retval true if the nonce is stale.
```

Comment:
Not adding this change because stale is a subset of invalid for nonce. Hopefully the update to clarify the preceeding comments will resolve this misunderstanding.

Reply:
Done in f9cf75b97444e0256c3c1d69520920a1a145fd3a